xrpl/models/transactions/xchain_claim.py

```diff
@@ -77,7 +78,17 @@ def _get_errors(self: Self) -> Dict[str, str]:
         errors = super()._get_errors()
 
         bridge = self.xchain_bridge
-        currency = XRP() if isinstance(self.amount, str) else self.amount.to_currency()
+
+        amount = self.amount
+        if is_xrp(amount):
+            currency = XRP()
+        elif is_issued_currency(amount):
+            currency = amount.to_currency()  # type: ignore
+        elif is_mpt(amount):
+            currency = amount.mpt_issuance_id  # type: ignore
+        else:
+            errors["amount"] = "currency can't be derived."
+
```

Comment:
why are these changes needed in python and not the JS lib?

Reply:
They're also in JS - https://github.com/XRPLF/xrpl.js/pull/2661/files#diff-695e6507bbf3a86833c7318c48da6ec25fe12c7a4f8a90a3a0cfbb8c19f8b3abR167